### PR TITLE
Fix false negative for calling module-level function before definition

### LINKED ISSUE #1144: Fail to find NameError for undefined functions
### Steps to reproduce

run pylint on code:

```
# coding=utf-8
print(calculate(1.01, 2))

def calculate(value1: int, value2: float) -> int:
    return (value1+value2)
```
### Current behavior

Error is not issued.
### Expected behavior

E:  nn, nn: Undefined function 'calculate' (undefined-function)
### pylint --version output

pylint.EXE 1.6.4,
astroid 1.4.8
Python 3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:43:06) [MSC v.1600 32 bit (Intel)]

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Closes #1144

### CODE REVIEW

**Review by DanielNoord — DISMISSED** [review]

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -1342,6 +1337,9 @@ def leave_module(self, node: nodes.Module) -> None:
 
     def visit_classdef(self, node: nodes.ClassDef) -> None:
         """Visit class: update consumption analysis variable."""
+        for name in node.nodes_of_class(nodes.AssignName):
+            # Names have to be inferred, because they could affect locals.
+            safe_infer(name)
```
Comment (jacobtylerwalls) [inline pylint/checkers/variables.py:1342]: This is quite lame, but we'll need PyCQA/astroid#1490 to get rid of it.

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: Worked around this.

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: ... but will still need an astroid update to pass the test suite. PR forthcoming.

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: See https://github.com/pylint-dev/astroid/pull/2119

Comment (Pierre-Sassoulas) [reply pylint/checkers/variables.py]: Should we backport the astroid update (even if this is used in a false negative that we won't backport) ?

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: Good question, but I'd rather not. It's not a very stable change.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM !

**Comment by jacobtylerwalls** [comment]
I was going to wait for a new primer result, but if there are follow-ups, I'll just open a new PR. Sorry for not saying something sooner!

**Comment by Pierre-Sassoulas** [comment]
That's my bad for merging too fast :)

**Comment by jacobtylerwalls** [comment]
No worries, I just checked and I can't reproduce the 2nd and 3rd items from the earlier primer comment, so I think they were also property-related like the 1st one I addressed in astroid (and checking the files confirms it) :tada:

**Pull request merged**